packages/ogm/src/utils/filter-document.ts

```diff
@@ -30,10 +30,20 @@ const excludedDirectives = [
     "private",
     "readonly",
     "writeonly",
+    "query",
+    "mutation",
+    "subscription",
+    "filterable",
+    "selectable",
+    "settable",
 ];
 
 function filterDocument(typeDefs: Neo4jGraphQLConstructor["typeDefs"]): DocumentNode {
-    const merged = mergeTypeDefs(Array.isArray(typeDefs) ? (typeDefs as string[]) : [typeDefs as string]);
+    // hack to keep aggregation enabled for OGM
+    const schemaExtension = "extend schema @query(aggregate: true)";
+    const merged = mergeTypeDefs(
+        Array.isArray(typeDefs) ? (typeDefs as string[]).concat(schemaExtension) : [typeDefs as string, schemaExtension]
+    );
```

Comment:
I see you add the schema extension in at this point - how does it not get removed by the reducer below? I would have thought it would have needed to be added at the end. Because if schema extensions aren't being filtered, then users will be able to add schema extensions also and potentially produce unexpected behaviour.

Reply:
Yeah, it's very hacky but if you see below only object types are transformed, so the extension definitions remain intact!

Reply:
Which is a problem right? Because it means people could pass in the `@mutation` directive for example and disable mutations, which would break the OGM. I think the reducer _does_ need to worry about schema extensions, and the extensions that we need should be added on following filtering.